decode.go

```diff
@@ -308,9 +308,10 @@ func (p *parser) mapping() *Node {
 // Decoder, unmarshals a node into a provided value.
 
 type decoder struct {
-	doc     *Node
-	aliases map[*Node]bool
-	terrors []string
+	doc           *Node
+	aliases       map[*Node]bool
+	terrors       []string
+	strictTerrors []string
```

Comment:
this should probably be `strictErrors`? I guess t stands for type, but it is still a bit weird when reading

Reply:
changed it